src/vm/database/clarity_db.rs

```diff
@@ -653,4 +782,43 @@ impl<'a> ClarityDatabase<'a> {
         let key = ClarityDatabase::make_key_for_account_nonce(principal);
         self.put(&key, &nonce);
     }
+
+    pub fn get_account_stx_locked(&mut self, principal: &PrincipalData) -> u128 {
```

Comment:
Should this data all just be stored in the same entry? By splitting them across different entries, it requires 3 MARF lookups per account query, which I think is a higher cost than the cost of duplicating the height and lock data on transfers.

Reply:
Sure, I can combine them.

Reply:
Addressed in https://github.com/blockstack/stacks-blockchain/pull/1850

Reply:
On second thought, we should not consolidate the balance with the block height and locked tokens.  This would unnecessarily penalize a token sender when it loads up the recipient's balance, because then the sender would also need to unnecessarily load the recipient's lock state and block height (even though they will not be used).  I think we could consolidate the block height and locked STX, though, since these values are always loaded and stored together.

Reply:
Discussion happening here - https://github.com/blockstack/stacks-blockchain/pull/1850#discussion_r486408181